cmd/object-handlers.go

```diff
@@ -48,16 +49,39 @@ func setGetRespHeaders(w http.ResponseWriter, reqParams url.Values) {
 	}
 }
 
+// getSourceIPAddress - get the source ip address of the request.
+func getSourceIPAddress(r *http.Request) string {
+	// Attempt to get ip from standard headers.
+	var ip string
+	for _, h := range []string{"X-Forwarded-For", "X-Real-Ip"} {
```

Comment:
Actually there is a problem if you directly use canonical strings for `X-Forwarded-For` It would be easier to just use r.Header.Get(h) instead of looping through multiple addresses..  It is also possible to spoof this information easily. 

>The general format of the field is:
>> X-Forwarded-For: client, proxy1, proxy2[3]

> where the value is a comma+space separated list of IP addresses, the left-most being the original client, and each successive proxy that passed the request adding the IP address where it received the request from. In this example, the request passed through proxy1, proxy2, and then proxy3 (not shown in the header). proxy3 appears as remote address of the request. Since it is easy to forge an X-Forwarded-For field the given information should be used with care. The last IP address is always the IP address that connects to the last proxy, which means it is the most reliable source of information. X-Forwarded-For data can be used in a forward or reverse proxy scenario. Just logging the X-Forwarded-For field is not always enough as the last proxy IP address in a chain is not contained within the X-Forwarded-For field, it is in the actual IP header. A web server should log BOTH the request's source IP address and the X-Forwarded-For field information for completeness.

Some more story here 
- https://serverfault.com/questions/314574/nginx-real-ip-header-and-x-forwarded-for-seems-wrong/414166#414166
- https://distinctplace.com/2014/04/23/story-behind-x-forwarded-for-and-x-real-ip-headers/

So from what it looks like `X-Forwarded-For` can be easily spoofed and will lead to allowing access to objects on malicious clients with this in mind i suggest we use `X-Real-Ip` only.  So it means that 

header.Get(`X-Real-Ip`) is all you need to do, since X-Real-Ip is only set by proxies and cannot be changed/spoofed by the client and are unset by the nginx at termination.